explainer-datamodel.md

```diff
@@ -0,0 +1,413 @@
+content) that overflows those bounds. `width` and `height` and corresponding `min-width`
+or `max-height` properties express the desired constraints for eventual line-box 
+computation when it comes to rendering the object model or returning metrics. Similarly,
+`overflow` and `clip-path` can further ensure the text content expands or is clipped to
+fit the desired constraints. These properties would only apply on the `FormattedText` 
+container object.
+
+CSS continues to evolve, and it makes sense to extend relevant new CSS properties to
+this object model as they become a part of the web platform. For example, while not
+widely supported at the time of writing, support for `shape-inside` (CSS Shapes L2)
+and CSS Exclusions provide exciting growth opportunities for text using this model.
+
+## WebIDL
+
+```webidl
+[Exposed=Window,Worker] 
+interface FormattedText { 
+  constructor();
+  constructor(DOMString textRunText...);
+  constructor(sequence<FormattedTextRunInit> textRunInit);
+  constructor(FormattedText copyConstructorInit);
+
+  attribute ObservableArray<FormattedTextRun> textruns; 
+}; 
+FormattedText includes FormattedTextStylable; 
+
+dictionary FormattedTextRunInit { 
+  DOMString text = ""; 
+  DOMString font = "";
+  DOMString color = "";
+}; 
+
+[Exposed=Window,Worker] 
+interface FormattedTextRun { 
+  constructor();
+  constructor(DOMString text);
+  constructor(FormattedTextRunInit textRunInit);
+
+  attribute DOMString text; 
+  attribute DOMString font; 
+  attribute DOMString color; 
+
+  // owning FormattedText (if any) 
+  readonly attribute FormattedText? owner;         // null if not contained in a FormattedText
+ 
+  // navigation in the array 
+  readonly attribute FormattedTextRun? next;       // null if last or not contained in FormattedText
+  readonly attribute FormattedTextRun? previous;   // null if first or not contained in FormattedText
+}; 
+FormattedTextRun includes FormattedTextStylable; 
+ 
+interface mixin FormattedTextStylable { 
+  [SameObject] readonly attribute StylePropertyMap styleMap;
+  void setStyle(USVString cssText);
+}; 
+```
+
+## Supported CSS on FormattedText and text runs
+
+We've compiled a list of text-related (or generally applicable) CSS properties that we believe make
+sense to support on `FormattedText` and `FormattedTextRun` objects. This list is not exhaustive.
+For example, it does not include many of the new logical properites such as `inline-size` for 
+`width`. This list is provided for potential testing purposes and to facilitate discussion.
+
+| CSS Property | FormattedText | FormattedTextRun | inherits | Notes |
+|--------------|---------------|------------------|----------|-------|
+| background | ✔ | ✔ | no | background-attachment, background-origin, background-position, background-repeat, background-size operate on an external resource image and will be ignored. Background-image will only support `<gradient>` functions |
```

Comment:
just FYI, adding background does not work today since we only paint the foreground layer. If we want to add background support we would need to define the drawing order which may be easy as claiming the FormattedText object is a full stacking context.

Reply:
Noted in https://github.com/WICG/canvas-formatted-text/commit/742d9b4c4c1f454ffe47383aafb803a169bdae98